package/MDAnalysis/core/Selection.py

```diff
@@ -178,6 +178,12 @@ def __init__(self, parser, tokens):
         pass
 
     def apply(self, group):
+        # Check whether group is identical to the one stored
+        # in the corresponding universe, in which case this
+        # is returned directly. This works since the Universe.atoms
+        # are unique by construction.
+        if group.universe and group is group.universe.atoms:
```

Comment:
I also think that @wouterboomsma 's code does what it is supposed to do.

I admit that I don't remember if we can really have AtomGroups that don't belong to a Universe. I don't think that this should happen under normal circumstances. In any case, for the future (I.e., #363) I defer to @richardjgowers and @dotsdl – they should know if this will be an issue for the future.